catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/event/SubscriptionManager.java

```diff
@@ -0,0 +1,345 @@
+                            e);
+                }
+
+                LOGGER.debug("Removing subscription from internal hash map");
+                registeredSubscriptions.remove(subscriptionId);
+                subscriptions.remove(subscriptionId);
+                LOGGER.info("Subscription removal complete");
+            } else {
+                LOGGER.debug("subscriptionId is NULL or empty");
+            }
+        } catch (Exception e) {
+            LOGGER.info("Could not delete subscription for " + subscriptionId);
+            LOGGER.error(e.getMessage(), e);
+        }
+
+        LOGGER.debug("EXITING: " + methodName + "    (status = " + status + ")");
+
+        return status;
+    }
+
+    /**
+     * Persist the subscription to the OSGi ConfigAdmin service. Persisted registeredSubscriptions can then be restored if DDF
+     * is restarted after a DDF outage or DDF is shutdown.
+     * Pass in client-provided subscriptionId and subscription UUID because the filter XML to be persisted for this subscription will be used to
+     * restore this subscription and should consist of the exact values the
+     * client originally provided.
+     */
+    protected static void persistSubscription(BundleContext context, Subscription subscription,
+            String deliveryMethodUrl, String subscriptionId, String subscriptionUuid) {
+        String methodName = "persistSubscription";
+        LOGGER.debug("ENTERING: " + methodName);
+
+        // Convert Subscription (which is an OGC Filter) to XML for persistence into OSGi CongiAdmin
+        String filterXml = convertSubscriptionToXml(subscription);
+
+        // Store filter XML, deliveryMethod URL, this endpoint's factory PID, and subscription ID into OSGi CongiAdmin
+        if (filterXml != null) {
+            try {
+                Configuration config = getConfigAdmin(context).createFactoryConfiguration(
+                        FACTORY_PID,
+                        null);
+
+                Dictionary<String, String> props = new Hashtable<String, String>();
+                props.put("subscriptionId", subscriptionId);
+                props.put("filterXml", filterXml);
+                props.put("deliveryMethodUrl", deliveryMethodUrl);
+                props.put("subscriptionUuid", subscriptionUuid);
+
+                LOGGER.debug("Done adding persisting subscription to ConfigAdmin");
+
+                config.update(props);
+            } catch (IOException e) {
+                LOGGER.warn("Unable to persist subscription " + subscriptionId, e);
+            }
+        }
+
+        LOGGER.debug("EXITING: " + methodName);
+    }
+
+    public static String convertSubscriptionToXml(Subscription subscription) {
+        String methodName = "convertSubscriptionToXml";
+        LOGGER.debug("ENTERING: " + methodName);
+        String filterXml = null;
+        // TODO (RWY) - figure out how to serialize the query from the subscription so it can be persisted.
```

Comment:
I have another ticket to work on persisting the subscription which I would probably move this logic out of here if possible and into the subscription so I will remove this method.